src/PlusDataCollection/RevopointSurface/vtkPlusRevopointSurface.cxx

```diff
@@ -0,0 +1,385 @@
+
+// Stl includes
+#include <vector>
+#include <string>
+#include <memory>
+#include <algorithm>
+
+//----------------------------------------------------------------------------
+vtkStandardNewMacro(vtkPlusRevopointSurface);
+
+namespace
+{
+  struct SurfaceStreamConfig
+  {
+    std::string Id;
+    StreamInfo StreamProps;
+    int DepthRange[2];
+    Intrinsics Intr;
+    Extrinsics Extr;
+    vtkPlusDataSource* Source{nullptr};
+  };
+
+  std::string convertErrorCode(ERROR_CODE error)
+  {
+    switch (error)
+    {
+      case ERROR_CODE::ERROR_PARAM:
+        return "Bad param";
+      case ERROR_CODE::ERROR_DEVICE_NOT_FOUND:
+        return "Device not found";
+      case ERROR_CODE::ERROR_DEVICE_NOT_CONNECT:
+        return "Device not connected";
+      case ERROR_CODE::ERROR_DEVICE_BUSY:
+        return "Device busy";
+      case ERROR_CODE::ERROR_STREAM_NOT_START:
+        return "Device not started";
+      case ERROR_CODE::ERROR_STREAM_BUSY:
+        return "Stream busy";
+      case ERROR_CODE::ERROR_FRAME_TIMEOUT:
+        return "Frame timeout";
+      case ERROR_CODE::ERROR_NOT_SUPPORT:
+        return "Not supported";
+      case ERROR_CODE::ERROR_PROPERTY_GET_FAILED:
+        return "Getting property failed";
+      case ERROR_CODE::ERROR_PROPERTY_SET_FAILED:
+        return "Setting property failed";
+      default:
+        break;
+    }
+    return "Success";
+  }
+}
+
+//----------------------------------------------------------------------------
+class vtkPlusRevopointSurface::vtkInternal
+{
+public:
+  vtkPlusRevopointSurface* External;
+  SurfaceStreamConfig DepthStream;
+  cs::ICameraPtr Camera;
+  cs::IFramePtr DepthFrame;
+
+  vtkInternal(vtkPlusRevopointSurface* external)
+    : External(external)
+  {
+    Camera = cs::getCameraPtr();
+  }
+
+  PlusStatus UpdateConfig()
+  {
+    std::vector<StreamInfo> streamProps;
+    Camera->getStreamInfos(STREAM_TYPE::STREAM_TYPE_DEPTH, streamProps);
+
+    bool foundMatchingConfig{false};
+    for (auto streamProp : streamProps)
+    {
+      LOG_TRACE("New stream configuration for format: " << streamProp.format);
+      if (streamProp.format == DepthStream.StreamProps.format)
+      {
+        LOG_INFO("New depth stream compliant configuration: " << streamProp.width << "x" << streamProp.height << "@" << streamProp.fps);
+        if (DepthStream.StreamProps.width != streamProp.width)
+        {
+          continue;
+        }
+
+        if (DepthStream.StreamProps.fps > streamProp.fps)
+        {
+          DepthStream.StreamProps.fps = streamProp.fps;
+          LOG_WARNING("Expected frame rate too high for current bandwidth. Actual framerate will be @" << streamProp.fps << " fps.");
+        }
+
+        DepthStream.StreamProps.height = streamProp.height;
+        foundMatchingConfig = true;
+        break;
+      }
+    }
+
+    if (!foundMatchingConfig)
+    {
+      LOG_ERROR("No matching configuration found. Check the manufacturer specifications.");
```

Comment:
If no matching configurations are found, we should log the list of supported configs.